modules/services/dnsmasq.nix

```diff
@@ -65,17 +65,24 @@ in
     };
   };
 
-  config = mkIf cfg.enable {
+  config = mkIf cfg.enable (let
+    addressArgs = mapA (domain: addr: "--address=/${domain}/${addr}") cfg.addresses;
+    serverArgs = map (server: "--server=${server}") cfg.servers;
+    dnsmasqArgs = [
+      "--listen-address=${cfg.bind}"
+      "--port=${toString cfg.port}"
+      "--keep-in-foreground"
+    ] ++ addressArgs ++ serverArgs;
+    dnsmasqCmd = "${cfg.package}/bin/dnsmasq ${concatStringsSep " " dnsmasqArgs}";
+  in {
     environment.systemPackages = [ cfg.package ];
 
     launchd.daemons.dnsmasq = {
       serviceConfig.ProgramArguments = [
```

Comment:
Instead of manually specifying `wait4path` you can just change this to `launchd.daemons.dnsmasq.command` like other nix-darwin services

Reply:
Done!